File: PhanDangNhanBI12-336 prac13/app/src/main/java/com/example/weather/WeatherPagerAdapter.java
package com.example.weather;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class WeatherPagerAdapter extends FragmentPagerAdapter {

    public WeatherPagerAdapter(@NonNull FragmentManager fm, int behavior) {
        super(fm, behavior);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        // Return a new instance of WeatherAndForecastFragment for each page
        return new WeatherAndForecastFragment();
    }

    @Override
    public int getCount() {
        // We want 3 pages (3 WeatherAndForecastFragments)
        return 3;
    }
}

File: PhanDangNhanBI12-336 prac13/USTHWeather/app/src/main/java/Adapter/WeatherPagerAdapter.kt
package Adapter

class WeatherPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getCount(): Int = 3

    override fun getItem(position: Int): Fragment = WeatherAndForecastFragment()

    override fun getPageTitle(position: Int): CharSequence {
        return when (position) {
            0 -> "Tab 1"
            1 -> "Tab 2"
            2 -> "Tab 3"
            else -> "Tab"
        }
    }
}
